plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/RetryingJdbcClient.java

```diff
@@ -0,0 +1,527 @@
+/*
```

Comment:
I will add this to commit message, but here we can have nested retrying. One in jdbc client and other in connection factory. I believe it is ok as in both places we are using retry policy with max duration to 30 seconds. So matter where we retry the 30s is the longest we would wait.